internal/server/pre_reboot.go

```diff
@@ -18,53 +18,58 @@ const (
 // PreReboot is the handler for the `/v1/pre-reboot` endpoint.
 func (a *Airlock) PreReboot() http.Handler {
 	handler := func(w http.ResponseWriter, req *http.Request) {
-		logrus.Debug("got pre-reboot request")
-		if a == nil {
-			http.Error(w, errNilAirlockServer.Error(), 500)
-			return
-		}
-
-		nodeIdentity, err := validateIdentity(req)
+		code, err := a.preRebootHandler(w, req)
 		if err != nil {
-			logrus.Errorln("failed to validate client identity: ", err)
-			http.Error(w, err.Error(), 400)
-			return
+			http.Error(w, err.Error(), code)
 		}
-		logrus.WithFields(logrus.Fields{
-			"group": nodeIdentity.Group,
-			"uuid":  nodeIdentity.UUID,
-		}).Debug("processing client pre-reboot request")
+	}
 
-		slots, ok := a.LockGroups[nodeIdentity.Group]
-		if !ok {
-			err := fmt.Errorf("unknown group %q", nodeIdentity.Group)
-			logrus.Errorln("unable to satisfy client request: ", err)
-			http.Error(w, err.Error(), 400)
-			return
-		}
+	return http.HandlerFunc(handler)
+}
 
-		ctx, cancel := context.WithTimeout(context.Background(), a.EtcdTxnTimeout)
-		defer cancel()
-		lockManager, err := lock.NewManager(ctx, a.EtcdEndpoints, nodeIdentity.Group, slots)
-		if err != nil {
-			logrus.Errorln("failed to initialize semaphore manager: ", err)
-			http.Error(w, err.Error(), 500)
-			return
-		}
-		defer lockManager.Close()
+// preRebootHandler contains pre-reboot handling logic
+func (a *Airlock) preRebootHandler(w http.ResponseWriter, req *http.Request) (int, error) {
```

Comment:
This function no longer uses the `http.ResponseWriter`, worth dropping it from the parameter list?

Reply:
I also added an explicit status-code write for the positive case, directly at the calling site.